components/Form/TextArea/index.js

```diff
@@ -1,9 +1,9 @@
-const TextArea = ({ label, name, onChange, value }) => {
+const TextArea = ({ label, name, onChange, value, datatestid }) => {
```

Comment:
Could be worthwhile passing through any "other" props to these components that needed updates in this PR - so that we can extend things more easily in future?

```suggestion
const TextArea = ({ label, name, onChange, value, ...others }) => {
```

And then later on used like...
```jsx
<div className="govuk-form-group" {...others}>
```

Reply:
Hm, so I think I didn't do it that way because the dash in the parameter name doesn't seem to be allowed (?). With .**..others** it'll end up being [**datatestid**=notes-input] and it would work but I'm not sure if that's ok

Reply:
Good point, it's not allowed usually for syntax reasons, like you say I can't create this object as it will throw a `SyntaxError` trying to subtract `testid` from `data` and then encountering a `:` instead of `=` or some other valid operation...
```js
{ data-testid: '123' }
```
We _can_ create an object with a string key though, something like this...
```js
const x = {};
x['data-testid'] = '123';
```
and this is pretty similar to what React is doing when it applies your props to a component - so for example this is a legitimate prop...
```jsx
<SomeComponent data-testid="123" />
```

So it is awkward to use dashes in properties that you are destructuring like here` { label, name, onChange, value, datatestid }`, when you do `...others` instead it will handle the awkwardness of the `-` being in the prop for you and you don't need to worry about it.

If you're interested, here's how you can do it manually...
```
const TextArea = (props) => {
   const { label, name, onChange } = props; // boring, no dash properties
   const testId = props['data-testid']; // exciting, prop with dash in the name
```

Anyway - this was a very long way of saying - you can use dashes in property names but it's quite awkward unless you know the tricks to do it. I'm cool with approving as-is, my concern is that it could get confusing which elements need `data-testid` and which need `datatestid` which could make it harder to maintain the project in the future - so I'd be keen to get this into the backlog to update soon if it's not done in this PR :)

Hope that makes sense, happy to chat if not!

Reply:
It makes sense! Thanks for explaining it, I'm totally for having it with a dash, so I'll fix it up now!

Reply:
Nice, thank you!

Reply:
Ended up removing it in general 🤔